xc/device.ml

```diff
@@ -1547,20 +1547,27 @@ let cmdline_of_disp info =
 		| GVT_d -> ["-std-vga"; "-gfx_passthru"]
 	in
 	let videoram_opt = ["-videoram"; string_of_int info.video_mib] in
+	let vnc_opts_of ip_addr_opt auto port keymap =
+		let combine_opts : bytes list option list -> bytes list = fun opts ->
+			List.map (function Some x -> x | _ -> []) opts
+			|> List.filter (function [] -> false | _ -> true)
+			|> List.flatten in
+		let ip_addr = Opt.default "127.0.0.1" ip_addr_opt in
+		let port = if auto then "1" else string_of_int port in
+		let unused_opt = if auto then Some ["-vncunused"] else None in
+		let vnc_opt = Some ["-vnc"; ip_addr ^ ":" ^ port] in
+		let keymap_opt = Some ["-k"; keymap] in
+		combine_opts [unused_opt; vnc_opt; keymap_opt]
+	in
```

Comment:
You can get the same effect without needing the `combine_opts` function:
```ocaml
let vnc_opts_of ip_addr_opt auto port keymap =
    let ip_addr = Opt.default "127.0.0.1" ip_addr_opt in
    let port = if auto then "1" else string_of_int port in
    let unused_opt = if auto then ["-vncunused"] else [] in
    let vnc_opt = ["-vnc"; ip_addr ^ ":" ^ port] in
    let keymap_opt = ["-k"; keymap] in
    List.flatten [unused_opt; vnc_opt; keymap_opt]
```
You should be able to make similar changes in other functions which are using `combine_opts`.